src/Zigzag_persistence/include/gudhi/Filtered_zigzag_persistence.h

```diff
@@ -0,0 +1,553 @@
+      }
+
+      if (death - birth > shortestInterval) {
+        diag.emplace_back(bar.dim(), birth, death);
+      }
+    }
+
+    return diag;
+  }
+
+  /**
+   * @brief Computes the births of the current essential cycles.
+   *
+   * @param diag Reference to vector where to store the infinit bars.
+   */
+  void _retrieve_infinit_bars(std::vector<Filtration_value_interval>& diag) {
+    auto birth = [this](internal_key birthKey) {
+      auto itBirth =  // lower_bound(x) returns leftmost y s.t. x <= y
+          std::lower_bound(
+              filtrationValues_.begin(), filtrationValues_.end(),
+              std::pair<internal_key, filtration_value>(birthKey, std::numeric_limits<filtration_value>::infinity()),
+              [](std::pair<internal_key, filtration_value> p1, std::pair<internal_key, filtration_value> p2) {
+                return p1.first < p2.first;
+              });
+      if (itBirth == filtrationValues_.end() || itBirth->first > birthKey) {
+        --itBirth;
+      }
+      return itBirth->second;
+    };
+
+    auto stream_infinit_interval = [&](dimension_type dim, internal_key birthIndex) {
+      if (dimMax_ == -1 || (dimMax_ != -1 && dim < dimMax_))
+        diag.emplace_back(dim, birth(birthIndex), std::numeric_limits<filtration_value>::infinity());
+    };
+
+    pers_.get_current_infinit_intervals(stream_infinit_interval);
+  }
+};  // end class Filtered_zigzag_persistence_with_storage
+
+/**
+ * @ingroup zigzag_persistence
+ *
+ * @brief Class computating the zigzag persistent homology of a zigzag filtration. Algorithm based on \cite zigzag.
+ * 
+ * @tparam FilteredZigzagOptions Structure following the @ref FilteredZigzagOptions concept.
+ * Default value: @ref Default_filtered_zigzag_options.
+ */
+template <class FilteredZigzagOptions = Default_filtered_zigzag_options>
+class Filtered_zigzag_persistence {
+ public:
+  using Options = FilteredZigzagOptions;                       /**< Zigzag options. */
+  using internal_key = typename Options::internal_key;         /**< Key and index type, has to be signed. */
+  using face_key = typename Options::face_key;                 /**< Face ID type from external inputs. */
+  using filtration_value = typename Options::filtration_value; /**< Type for filtration values. */
+  using dimension_type = typename Options::dimension_type;     /**< Type for dimension values. */
+
+  /**
+   * @brief Constructor.
+   * @details After construction of the class, the zigzag filtration should be given in a streaming like way, i.e.,
+   * call @ref insert_face, @ref remove_face or @ref apply_identity for each step of the filtration in order of
+   * the filtration. The bars of the diagram are retrieved via the given callback method every time
+   * a pair with non-zero length is closed. To retrieve the open/infinit bars, use @ref get_current_infinit_intervals.
+   *
+   * @param stream_interval Callback method to process the birth and death values of a persistence bar.
```

Comment:
I am trying to think if some users might care about more information about an interval. For instance, when you look at the interlevelset persistence, some interval extremities move up-left in the pyramid, while others move up-right, depending if the event was an insertion or a removal. If the user needs this information when processing an interval, I think they will need to use the unfiltered zigzag, maintain their own filtration map, and maintain a set to remember what operations were insertions.